onnxruntime/core/providers/tensorrt/tensorrt_execution_provider_utils.h

```diff
@@ -222,13 +222,28 @@ class TRTModelIdGenerator {
     if (!model_path.IsEmpty()) {
       // Get model name
       PathString path_string = model_path.GetComponents().back();
-      char arr[256];
+
 #ifdef _WIN32
-      wcstombs_s(nullptr, arr, sizeof(arr), path_string.c_str(), sizeof(arr));
+      std::string model_name;
+
+      // Convert wide model name to multibyte characters.
+      {
+        // Query the required buffer size (including '\0') by passing nullptr for the destination.
+        size_t total_bytes = 0;
+        ORT_IGNORE_RETURN_VALUE(wcstombs_s(&total_bytes, nullptr, 0, path_string.c_str(), _TRUNCATE));
+
+        // Resize the destination buffer and convert.
+        model_name.resize(total_bytes);
+        auto err = wcstombs_s(&total_bytes, &model_name[0], total_bytes, path_string.c_str(), _TRUNCATE);
+        ORT_ENFORCE(err == 0, MakeString(
+            "TRTGenerateId: Failed to convert wide model name to bytes. wcstombs_s returned ", err));
+
+        model_name.resize(total_bytes - 1);  // Remove the duplicate terminating '\0'
+      }
 #else
-      strcpy(arr, path_string.c_str());
+      std::string model_name(path_string);
 #endif
-      std::string model_name(arr);
+
       LOGS_DEFAULT(INFO) << "[TensorRT EP] Model name is " << model_name;
       // Ensure enough characters are hashed in case model names are too short
       int32_t model_name_length = gsl::narrow_cast<int32_t>(model_name.size());
```

Comment:
There's no need for this to be a `int32_t`. We can use `size_t` and avoid the need for a narrow cast.